src/main/java/it/gov/pagopa/payhub/activities/service/treasury/TreasuryOpiParserService.java

```diff
@@ -36,13 +58,38 @@ public List<String> parseData(Path ingestionFlowFilePath) {
             try {
                 flussoGiornaleDiCassa14 = treasuryUnmarshallerService.unmarshalOpi14(ingestionFlowFile);
                 log.debug("file flussoGiornaleDiCassa with Id {} parsed successfully ", flussoGiornaleDiCassa14.getId());
+                versione = TreasuryValidatorService.V_14;
             } catch (Exception exception) {
                 log.info("file flussoGiornaleDiCassa parsing error with opi 1.4 format {} ", exception.getMessage());
                 throw new TreasuryOpiInvalidFileException("Cannot parse treasury Opi file " + ingestionFlowFile);
             }
         }
-        //TODO in task 1658 it will be implemented the element to be returned
-        return List.of();
+
+
+        assert versione != null;
```

Comment:
let's not use `assert` to throw an errore if a condition fails!
assertions could be disabled, let's just throw an exception...

Anyway, version cannot be null! you could simply remove this line

Reply:
removed assert